File: 1392. Find the Difference of Two Arrays/Solution.py
class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        freq1 = {}
        freq2 = {}
        
        for num in nums1:
            freq1[num] = freq1.get(num, 0) + 1
        
        for num in nums2:
            freq2[num] = freq2.get(num, 0) + 1
        
        difference1 = []
        difference2 = []
        
        for num in freq1:
            if num not in freq2:
                difference1.append(num)
        
        for num in freq2:
            if num not in freq1:
                difference2.append(num)
        
        return [difference1, difference2]

File: 1392. Find the Difference of Two Arrays/Solution.cpp
#include <vector>
#include <unordered_set>

class Solution {
public:
    std::vector<std::vector<int>> findDifference(std::vector<int>& nums1, std::vector<int>& nums2) {
        std::unordered_set<int> set1(nums1.begin(), nums1.end());
        std::unordered_set<int> set2(nums2.begin(), nums2.end());
        
        std::vector<int> difference1;
        std::vector<int> difference2;
        
        for (int num : set1) {
            if (set2.find(num) == set2.end()) {
                difference1.push_back(num);
            }
        }
        
        for (int num : set2) {
            if (set1.find(num) == set1.end()) {
                difference2.push_back(num);
            }
        }
        
        return {difference1, difference2};
    }
};
